include/boost/geometry/algorithms/detail/closest_points/linear_or_areal_to_areal.hpp

```diff
@@ -0,0 +1,269 @@
+    static inline void apply(Areal const& areal,
+                             Linear const& linear,
+                             Segment& shortest_seg,
+                             Strategies const& strategies)
+    {
+        linear_to_areal::apply(linear, areal, shortest_seg, strategies);
+        detail::closest_points::swap_segment_points::apply(shortest_seg);
+    }
+};
+
+struct segment_to_areal
+{
+    template <typename Segment, typename Areal, typename OutSegment, typename Strategies>
+    static inline void apply(Segment const& segment,
+                             Areal const& areal,
+                             OutSegment& shortest_seg,                       
+                             Strategies const& strategies,
+                             bool = false)
+    {
+        using linestring_type = geometry::model::linestring
+            <typename point_type<Segment>::type>;
+        linestring_type linestring;
+        convert(segment, linestring);
+        linear_to_areal::apply(linestring, areal, shortest_seg, strategies);
+    }
+};
+
+struct areal_to_segment
+{
+    template <typename Areal, typename Segment, typename OutSegment, typename Strategies>
+    static inline void apply(Areal const& areal,
+                             Segment const& segment,
+                             OutSegment& shortest_seg,                       
+                             Strategies const& strategies,
+                             bool = false)
+    {
+        segment_to_areal::apply(segment, areal, shortest_seg, strategies);
+        detail::closest_points::swap_segment_points::apply(shortest_seg);
+    }
+};
+
+struct areal_to_areal
+{
+    template <typename Areal1, typename Areal2, typename Segment, typename Strategies>
+    static inline void apply(Areal1 const& areal1,
+                             Areal2 const& areal2,
+                             Segment& shortest_seg,
+                             Strategies const& strategies)
+    {
+        using most_precise_type = typename select_coordinate_type
+                <
+                    Areal1,
+                    Areal2
+                >::type;
+
+        using point_type = typename std::conditional
+        <
+            std::is_same<coordinate_type<Areal1>, most_precise_type>::value,
+            typename point_type<Areal1>::type,
+            typename point_type<Areal2>::type
+        >::type;
+
+        using linestring_type = geometry::model::linestring<point_type>;
+        using polygon_type = geometry::model::polygon<point_type>;
+        
+        /* TODO: currently intersection does not support tupled input
+         *       this should be implemented directly with dynamic geometries
+        */
+            
+        geometry::model::multi_point<point_type> mp_out;
+        geometry::intersection(areal1, areal2, mp_out, strategies);
+        
+        if (! boost::empty(mp_out))
+        {
+            set_segment_from_points::apply(*boost::begin(mp_out), 
```

Comment:
In case you wanted to make this shorter we have `util::front()`.

Reply:
sure, but I do not want to assert on empty `mp_out`

Reply:
`mp_out` is checked for emptiness above so the assertion will not fail. Or do you mean that you don't want to perform an additional unnecessary check? Anyway I'm ok with leaving it as it is.

Reply:
> Or do you mean that you don't want to perform an additional unnecessary check?

yes